crates/hir-ty/src/consteval.rs

```diff
@@ -248,17 +248,19 @@ pub(crate) fn const_eval_query(
     subst: Substitution,
     trait_env: Option<Arc<TraitEnvironment>>,
 ) -> Result<Const, ConstEvalError> {
-    let body = match def {
-        GeneralConstId::ConstId(c) => {
-            db.monomorphized_mir_body(c.into(), subst, db.trait_environment(c.into()))?
-        }
-        GeneralConstId::StaticId(s) => {
-            let krate = s.module(db).krate();
-            db.monomorphized_mir_body(s.into(), subst, TraitEnvironment::empty(krate))?
-        }
-    };
-    let c = interpret_mir(db, body, false, trait_env)?.0?;
-    Ok(c)
+    crate::next_solver::tls::with_db(db, || {
```

Comment:
I didn't follow the conversations on zulip too much but this seems weird. We are inside a query so why do we need to set the tls here manually. We should be able to just use salsa's attached thread local for this purpose https://github.com/salsa-rs/salsa/blob/master/src/attach.rs where required